### PR TITLE
New functions added to obtain reviews by their IDs

### PR DESCRIPTION
**Pull request by JLaw96**
Functions added to models/controllers to obtain review by ID and tested for 404 errors in the app.test file. The errors have also been handled in the error controllers file

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -134,4 +134,37 @@ describe("app", () => {
...
     });
   });
+  describe("GET /api/reviews/:review_id", () => {
+    it("200: GET - This test should respond with the data for the specific review that has been requested in the path (in this case review 1)", () => {
+      return request(app)
+        .get("/api/reviews/1")
+        .expect(200)
+        .then(({ body }) => {
+          const { reviews } = body;
+          expect(reviews).toEqual({
```
Comment (Augs0) [inline __tests__/app.test.js]: A better matcher here would be toMatchObject or objectContaining to allow us to expand this test later :)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -134,4 +134,37 @@ describe("app", () => {
...
+    it("404: GET - This test should return a 404 error with the correct message as our path/request contains a review_id that does not exist", () => {
+      return request(app)
+        .get("/api/reviews/500")
+        .expect(404)
+        .then(({ body }) => {
+          console.log(body, "log in test file");
+          expect(body).toHaveProperty("message", "review_id not found");
+        });
+    });
+  });
```
Comment (Augs0) [inline __tests__/app.test.js:167]: We also need a 400 test here for if the id is a string or something like that (not a number). PSQL will trigger an error if that's the case, so have a think how we could use our error handler

**Review by Augs0 — CHANGES_REQUESTED** [review]
Almost there Johnny just a few extra things to add :)

**Review by JVal-NC — COMMENTED** [review]
Looks good, go ahead and merge

**Pull request merged**